src/main/java/uk/gov/pay/api/resources/PaymentsResource.java

```diff
@@ -241,7 +247,11 @@ public Response searchPayments(@ApiParam(value = "accountId", hidden = true)
             @ApiResponse(code = 429, message = "Too many requests", response = ApiErrorResponse.class),
             @ApiResponse(code = 500, message = "Downstream system error", response = PaymentError.class)})
     public Response createNewPayment(@ApiParam(value = "accountId", hidden = true) @Auth Account account,
-                                     @ApiParam(value = "requestPayload", required = true) ValidCreatePaymentRequest validCreatePaymentRequest) {
+                                     @ApiParam(value = "requestPayload", required = true) CreatePaymentRequest createPaymentRequest) {
+        logger.info("Payment create request parsed to {}", createPaymentRequest);
+
+        paymentRequestValidator.validate(createPaymentRequest);
+        ValidCreatePaymentRequest validCreatePaymentRequest = new ValidCreatePaymentRequest(createPaymentRequest);
```

Comment:
Out of interest, have we decided yet if we’re planning to keep `ValidCreatePaymentRequest` in the long term? On the one hand, it’s great to have specific type for a validated request (it’s almost [typestate](https://en.wikipedia.org/wiki/Typestate_analysis)!) but it might be a bit redundant when we have `@Valid` and can basically assume that any request that survived deserialisation into a `CreatePaymentRequest` is valid.

Reply:
Happy to discuss, my intention was to remove it since as you say its superfluous once we have validation on  `CreatePaymentRequest`, it also follows our pattern in Connector to have just one request type which is validated rather than parsing into a second.

Reply:
Yeah, that sounds reasonable.